lib/errors.js

```diff
@@ -0,0 +1,74 @@
+class StorageEmpty extends Error {
+  constructor () {
+    super('No Hypercore is stored here')
+    this.code = 'STORAGE_EMPTY'
+  }
+}
+
+class StorageConflict extends Error {
+  constructor () {
+    super('Another Hypercore is stored here')
+    this.code = 'STORAGE_CONFLICT'
+  }
+}
+
+class InvalidSignature extends Error {
+  constructor () {
+    super('Remote sent invalid signature')
+    this.code = 'INVALID_SIGNATURE'
+  }
+}
+
+class InvalidCapability extends Error {
+  constructor () {
+    super('Remote sent invalid capability')
+    this.code = 'INVALID_CAPABILITY'
+  }
+}
+
+class SnapshotNotAvailable extends Error {
+  constructor () {
+    super('Snapshot is not available')
+    this.code = 'NOT_AVAILABLE'
```

Comment:
There are reason for `NOT_AVAILABLE` instead of `SNAPSHOT_NOT_AVAILABLE`? Idea that `BlockNotAvailable` will all use the same code? If not, should be explicit here

Reply:
Yes, was thinking other things would reuse this code

Reply:
... but I'm down to simplify ie not future proof

Reply:
(I convinced to myself to rename this one)